### DISCUSSION TITLE
Direct Connect requirements?

### DISCUSSION THREAD

**Discussion by AlmightyJojo** [rustdesk/rustdesk#4631] [General]
Hi - I've reviewed https://github.com/rustdesk/rustdesk/wiki/How-does-RustDesk-work%3F
I do not understand why relay seems to be forced. Always relay = N
Can someone help explain the requirement for TCP hole punching to succeed? 

Environment:
Self Hosted w/ nightly client 1.2
DMZ net = rustdesk server (Debian - docker)
LAN net = Client PC (A) & Remote PC (B)
21115-21119 tcp , 21116 udp opened up on switch ACLs.
Split DNS [rustdesk.mydomain.com](https://rustdesk.mydomain.com:21117/) resolves local DMZ IP.

rCURRENT.log
rendezvous server: [rustdesk.mydomain.com:21116](https://rustdesk.mydomain.com:21116/)
#1 punch attempt with 10.x.x.x:49162, id: 3562108235
relay requested from peer, time used: 2.6536ms, relay_server: [rustdesk.mydomain.com:21117](https://rustdesk.mydomain.com:21117/)

Tried temporarily disabling remote PC Win firewall. Always relayed.

**Comment by Cabus1982** [comment]
Same

**Comment by Zixim** [comment]
same...
We really need a troubleshooting guide on hole punching.
